File: cn/java-spring-demo/src/main/java/com/itangcent/springboot/demo/controller/FileCtrl.java
package com.itangcent.springboot.demo.controller;

import com.itangcent.springboot.demo.common.dto.IResult;
import com.itangcent.springboot.demo.common.dto.Result;
import com.itangcent.springboot.demo.common.dto.UserDto;
import com.itangcent.springboot.demo.common.model.UserInfo;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.multipart.MultipartFile;

/**
 * 文件上传控制器
 * 处理用户相关的文件上传操作，包括单文件和多文件上传功能。
 * 支持用户信息和文件的同时提交，以及分离提交两种方式。
 *
 * @module user
 */
@RestController
@RequestMapping(value = "file")
public class FileCtrl {
    /**
     * 创建新用户并上传头像
     * 使用@ModelAttribute接收表单数据和文件，实现用户信息和头像的一次性提交。
     * 
     * @param userDto 包含用户信息和头像文件的数据传输对象 - UserDto中包含用户基本信息和MultipartFile类型的头像文件
     * @return 返回保存结果，成功时返回"保存成功"消息
     */
    @PostMapping("/add")
    public IResult add(@ModelAttribute UserDto userDto) {
        return Result.success("保存成功");
    }

    /**
     * 创建新用户并上传头像（分离参数方式）
     * 用户信息和文件分开接收，提供更灵活的提交方式
     *
     * @param userInfo 用户基本信息 - 包含用户名、年龄等基本信息的对象
     * @param profileImg 用户头像文件 - 单个头像文件，使用MultipartFile接收
     * @return 返回保存结果，成功时返回"保存成功"消息
     * @undone 待完善的接口 - 需要添加文件大小验证和类型验证
     */
    @PostMapping("/add2")
    public IResult add(UserInfo userInfo,
                       @RequestParam(name = "profileImg") MultipartFile profileImg) {
        return Result.success("保存成功");
    }

    /**
     * 创建新用户并上传多个头像
     * 支持批量上传头像文件，可以同时上传多张头像图片
     *
     * @param userInfo 用户基本信息 - 包含用户名、年龄等基本信息的对象
     * @param profileImgs 多个头像文件数组 - 支持同时上传多个头像文件
     * @return 返回保存结果，成功时返回"保存成功"消息
     * @undone 待完善的接口 - 需要添加文件数量限制和总大小限制
     */
    @PostMapping("/add3")
    public IResult add(UserInfo userInfo,
                       @RequestParam(name = "profileImg") MultipartFile[] profileImgs) {
        return Result.success("保存成功");
    }
}

File: en/kotlin-spring-demo/src/main/kotlin/com/itangcent/springboot/kotlin/demo/controller/FileCtrl.kt
package com.itangcent.springboot.kotlin.demo.controller

import com.itangcent.common.dto.IResult
import com.itangcent.common.dto.Result
import com.itangcent.common.dto.UserDto
import com.itangcent.common.model.UserInfo
import org.springframework.web.bind.annotation.*
import org.springframework.web.multipart.MultipartFile

/**
 * File Upload Controller
 * Provides file upload functionality for user-related operations, such as avatar uploads
 *
 * @module user
 */
@RestController
@RequestMapping(value = ["/file"])
class FileCtrl {
    /**
     * Add New User (Including Avatar Upload)
     * Uses ModelAttribute to receive user information and avatar file
     *
     * @param userDto Data transfer object containing user information and avatar file
     * @return Returns save result message
     */
    @PostMapping("/add")
    fun add(@ModelAttribute userDto: UserDto): IResult<String> {
        return Result.success("Save successful")
    }

    /**
     * Add New User (Separate Avatar Upload)
     * Demonstrates how to receive user information and avatar file separately
     *
     * @param userInfo Basic user information
     * @param profileImg User avatar file
     * @return Returns save result message
     * @undone To be implemented
     */
    @PostMapping("/add2")
    fun add(userInfo: UserInfo,
            @RequestParam(name = "profileImg") profileImg: MultipartFile): IResult<String> {
        return Result.success("Save successful")
    }

    /**
     * Add New User (Support Multiple Avatars)
     * Demonstrates how to handle multiple file upload scenarios
     *
     * @param userInfo Basic user information
     * @param profileImgs Array of multiple avatar files
     * @return Returns save result message
     * @undone To be implemented
     */
    @PostMapping("/add3")
    fun add(userInfo: UserInfo,
            @RequestParam(name = "profileImg") profileImgs: Array<MultipartFile>): IResult<String> {
        return Result.success("Save successful")
    }
}